logging/logging_test.go

```diff
@@ -58,118 +52,98 @@ func init() {
 	log.SetOutput(os.Stdout)
 }
 
-/*******************************************************************************
+/***********************************************************************************************************************
  * Types
- ******************************************************************************/
+ **********************************************************************************************************************/
 
-type testServiceProvider struct {
-	services map[string]*launcher.Service
+type testInstanceIDProvider struct {
+	instances map[string]cloudprotocol.InstanceFilter
 }
 
-/*******************************************************************************
- * Vars
- ******************************************************************************/
-
-var systemd *dbus.Conn
-
-var serviceProvider = testServiceProvider{services: make(map[string]*launcher.Service)}
-
-/*******************************************************************************
- * Main
- ******************************************************************************/
-
-func TestMain(m *testing.M) {
-	if err := setup(); err != nil {
-		log.Fatalf("Error setting up: %s", err)
-	}
-
-	ret := m.Run()
-
-	cleanup()
-
-	os.Exit(ret)
+type testSystemdJournal struct {
+	sync.RWMutex
+	messages       []*sdjournal.JournalEntry
+	currentMessage int
+	systemdMatches []string
+	simulateError  bool
 }
 
 /*******************************************************************************
  * Tests
  ******************************************************************************/
 
 func TestGetServiceLog(t *testing.T) {
+	instanceProvider := testInstanceIDProvider{instances: make(map[string]cloudprotocol.InstanceFilter)}
+	defer instanceProvider.Close()
+
+	testJournal := testSystemdJournal{}
+	logging.SDJournal = &testJournal
+
 	logging, err := logging.New(&config.Config{Logging: config.Logging{
 		MaxPartSize: 1024, MaxPartCount: 10,
-	}}, &serviceProvider)
+	}}, &instanceProvider)
 	if err != nil {
 		t.Fatalf("Can't create logging: %s", err)
 	}
 	defer logging.Close()
 
 	from := time.Now()
 
-	if err = createService("logservice0"); err != nil {
-		t.Fatalf("Can't create service: %s", err)
-	}
+	subject := "subject0"
 
-	if err = startService("logservice0"); err != nil {
-		t.Fatalf("Can't start service: %s", err)
-	}
+	instance := uint64(0)
 
-	time.Sleep(10 * time.Second)
+	instanceFilter := cloudprotocol.InstanceFilter{ServiceID: "logservice0", SubjectID: &subject, Instance: &instance}
 
```

Comment:
Create separate function to fill instanceFilter. Or you can create functions to create string and unit pointers and use here in place:
```golang
func newString(value string) *string {
     return &value
}

cloudprotocol.InstanceFilter {SubjectID: newString("subject0")
```
```

Reply:
added createInstanceFilter method